ecl/eclagent/eclagent.cpp

```diff
@@ -1650,9 +1650,44 @@ IHThorGraphResults * EclAgent::createGraphLoopResults()
 
 //---------------------------------------------------------------------------
 
-void addException(IWorkUnit *w, ErrorSeverity severity, const char * source, unsigned code, const char * text, const char * filename, unsigned lineno, unsigned column, bool failOnError)
+void addException(IWorkUnit *w, ErrorSeverity severity, MessageAudience aud, const char * source, unsigned code, const char * text, const char * filename, unsigned lineno, unsigned column, bool failOnError)
 {
-    UERRLOG("%s", text);
+    if (severity==SeverityInformation || severity==SeverityWarning)
+    {
+        switch (aud)
+        {
+            case MSGAUD_operator:
+                OERRLOG("%s", text);
+                break;
+            case MSGAUD_user:
+                UERRLOG("%s", text);
+                break;
+            case MSGAUD_audit:
+                AERRLOG("%s", text);
+                break;
+            default:
+                IERRLOG("%s", text);
+                break;
+        }
+    }
+    else
+    {
+        switch (aud)
+        {
+            case MSGAUD_operator:
+                OERRLOG("%s", text);
+                break;
+            case MSGAUD_user:
+                UERRLOG("%s", text);
+                break;
+            case MSGAUD_audit:
+                AERRLOG("%s", text);
+                break;
+            default:
+                IERRLOG("%s", text);
+                break;
+        }
+    }
```

Comment:
What is the difference between these switch blocks ?  They seem identical and severity is not used in either of them ??

Reply:
The two block shouldn't have been the same.  I've corrected (now in a separate function for re-use)